### PR TITLE
Add more integration tests and fix issues

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

**Comment by matriv** [comment]
@seut Please review

File: sql/src/main/java/io/crate/planner/consumer/GlobalAggregateConsumer.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ private static Plan globalAggregates(Functions functions,
             plannerContext.nextExecutionPhaseId(),
             "mergeOnHandler",
             collectPhase.nodeIds().size(),
-            Collections.emptyList(),
+            Collections.singletonList(plannerContext.handlerNode()),
```
Comment (seut) [inline sql/src/main/java/io/crate/planner/consumer/GlobalAggregateConsumer.java]: shouldn't be needed anymore?

File: sql/src/main/java/io/crate/planner/consumer/NonDistributedGroupByConsumer.java [hunk 2]
```diff
@@ -168,15 +173,19 @@ private Plan nonDistributedGroupBy(QueriedTableRelation table,
...
-            return Merge.create(
-                collect,
-                context.plannerContext(),
+
+            MergePhase mergePhase = new MergePhase(
+                plannerContext.jobId(),
+                plannerContext.nextExecutionPhaseId(),
+                "mergeOnHandler",
+                collect.resultDescription().nodeIds().size(),
+                Collections.singletonList(plannerContext.handlerNode()),
```
Comment (seut) [inline sql/src/main/java/io/crate/planner/consumer/NonDistributedGroupByConsumer.java]: here also, do we need to set the (handler) nodes here?

File: sql/src/main/java/io/crate/planner/node/dql/RoutedCollectPhase.java [hunk 3]
```diff
@@ -306,6 +306,7 @@ public RoutedCollectPhase normalize(EvaluatingNormalizer normalizer, Transaction
                 newWhereClause,
                 distributionInfo
             );
+            result.orderBy(orderBy);
```
Comment (seut) [inline sql/src/main/java/io/crate/planner/node/dql/RoutedCollectPhase.java]: afaik we should patch this against master. also I think it was not related to unions at all but just occurred while developing. would be also interesting why this never caused issues on master

File: sql/src/main/java/io/crate/planner/projection/TopNProjection.java [hunk 4]
```diff
@@ -119,10 +121,13 @@ public boolean isOrdered() {
 
     @Override
     public void prependOutput(Symbol symbol) {
-        InputColumn.shiftRight(outputs);
-        outputs.add(0, symbol);
+        Set<Symbol> symbolsToShift = Sets.newHashSet(outputs);
```
Comment (seut) [inline sql/src/main/java/io/crate/planner/projection/TopNProjection.java]: pls move down to where it is used. btw. afaik I assigned a local var here by intent for clarification

File: sql/src/main/java/io/crate/analyze/symbol/InputColumn.java [hunk 5]
```diff
@@ -153,6 +141,7 @@ public boolean equals(Object o) {
         InputColumn that = (InputColumn) o;
 
         if (index != that.index) return false;
+        if (!dataType.equals(that.dataType)) return false;
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/symbol/InputColumn.java]: Without it:
`java.lang.AssertionError: 
Expected: is "2| Rupert\n2| Ford\n1| Earth\n1| Arthur\n"
     but: was "2| 12884901889\n2| 4294967297\n1| 12884901888\n1| 4294967296\n" <Click to see difference>


	at __randomizedtesting.SeedInfo.seed([C70ED727E423FA9E:E913CD112D203A42]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at io.crate.integrationtests.FetchOperationIntegrationTest.testUnionFetch(FetchOperationIntegrationTest.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1764)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:871)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:907)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:921)`

File: sql/src/main/java/io/crate/executor/transport/TransportExecutor.java [hunk 6]
```diff
@@ -207,11 +208,13 @@ public Task visitDropTablePlan(DropTablePlan plan, Void context) {
 
         @Override
         public Task visitKillPlan(KillPlan killPlan, Void context) {
-            return killPlan.jobToKill().isPresent() ?
-                new KillJobTask(transportActionProvider.transportKillJobsNodeAction(),
-                    killPlan.jobId(),
-                    killPlan.jobToKill().get()) :
-                new KillTask(transportActionProvider.transportKillAllNodeAction(), killPlan.jobId());
+            Optional<UUID> jobToKill = killPlan.jobToKill();
+            return jobToKill.isPresent() ?
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/executor/transport/TransportExecutor.java]: can you avoid doing unrelated style changes like this in this PR?

**Comment by matriv** [comment]
New PR will be opened
